.github/workflows/lint-and-test.yml

```diff
@@ -0,0 +1,49 @@
+name: 'Linting and Testing'
+
+on: pull_request
+
+concurrency:
+  group: lint-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  run-tests:
+    name: 'Run Tests'
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Node.js and use yarn
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: 'yarn'
+
+      - name: Install All Dependencies
+        run: yarn
+
+      - name: Check Formatting with Prettier
+        run: yarn format-check-all
+
+      - name: Lint JS with ESLint
+        run: yarn lint
+
+      - name: Lint CSS with Stylelint
+        run: yarn lint-css
+
+      - name: Check TypeScript compilation
+        run: yarn lint-ts
+
+      - name: Run tests
+        run: yarn test
+
+      # - name: Publish to Chromatic
+      #   uses: chromaui/action@v1
+      #   with:
+      #     token: ${{ secrets.GITHUB_TOKEN }}
+      #     projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+      #     exitOnceUploaded: true
```

Comment:
Features of the new workflow are cobbled together from all the different GHA lint workflows

- `concurrency` cancels old runs when new ones on the same PR supercede them
- All linters are run in this order: `prettier`, `eslint`, `stylelint`, `tsc`, `yarn test`, then deploy to Chromatic. Any steps that aren't applicable are currently commented out. The action should effectively "fail-fast" at the earliest failed step, which has benefits in minimizing cycles per run but could cause issues if there are errors in later linters that require extra iterations to fix (i.e. the pusher is depending on CI instead of running it themselves).
- Using new command names that match each repo.

Reply:
> Any steps that aren't applicable are currently commented out.

I think we can simply remove them.

Reply:
Comments are removed and will be in the mirrored PRs